pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -455,6 +459,36 @@ private void handlePostCreation()
     LOGGER.info("Driver, indexing time : {}", totalIndexTime);
   }
 
+  private void updateSegmentStartEndTimeIfNecessary(ColumnStatistics timeColumnStats) {
+    switch (config.getTimeColumnType()) {
+      case EPOCH:
+        break;
+      case SIMPLE_DATE:
+        long startTime = convertStartTimeSDFToMillis(timeColumnStats);
+        config.getCustomProperties().put(V1Constants.MetadataKeys.Segment.SEGMENT_START_TIME, String.valueOf(startTime));
+        long endTime = convertEndTimeSDFToMillis(timeColumnStats);
+        config.getCustomProperties().put(V1Constants.MetadataKeys.Segment.SEGMENT_END_TIME, String.valueOf(endTime));
+        break;
+    }
+  }
+
+  public long convertStartTimeSDFToMillis(ColumnStatistics timeColumnStats) {
+    final String minTimeStr = timeColumnStats.getMinValue().toString();
+    return convertSDFToMillis(minTimeStr);
+  }
+
+  public long convertEndTimeSDFToMillis(ColumnStatistics timeColumnStats) {
+    final String maxTimeStr = timeColumnStats.getMaxValue().toString();
+    return convertSDFToMillis(maxTimeStr);
+  }
+
+  private long convertSDFToMillis(final String colValue) {
+    final String sdfFormatStr = config.getSimpleDateFormat();
```

Comment:
protect here. format can be null (bad coding) and colValue can be empty if there is no time column which will trigger exception